stripped/colabs/GrupowanieEM_stripped.py

```diff
@@ -0,0 +1,396 @@
+
+NASA2020 = NASA2020.merge(NASA2020_median, how='inner', on=['lon', 'lat'])
+
+
+for i in range(4, 10):
+    NASA2020 = add_column('RootMoist', i , NASA2020_full, NASA2020)
+
+for i in range(4, 11):
+    NASA2020 = add_column('GVEG', i , NASA2020_full, NASA2020)
+
+for feat in ['Rainf', 'PotEvap', 'Evap', 'Albedo']:
+   for i in range(5, 10):
+
+     NASA2020 = add_column(feat, i, NASA2020_full, NASA2020)
+
+
+NASA2020.head()
+
+NASA2020feat = NASA2020.drop(columns=['lon', 'lat'])
+
+#standaryzujemy dane:
+scaler = preprocessing.StandardScaler()
+scaler.fit(NASA2020feat)
+standarized = scaler.transform(NASA2020feat)
+
+st_NASA2020 = pd.DataFrame(standarized, columns=NASA2020feat.columns)
+st_NASA2020.head()
+
+gm = GaussianMixture(n_components = 2, n_init = 200, max_iter=200, init_params= 'random_from_data', covariance_type='spherical', random_state=22)
+gm_result = gm.fit_predict(st_NASA2020)
+
+gm_list = gm_result.tolist()
+number_of_label1 = sum(gm_list)
+print(number_of_label1)
+
+percentage_deserts_predicted = sum(gm_list)/len(NASA2020)
+print(percentage_deserts_predicted)
+
+NASA2020['label'] = gm_list
+anotated_unique = NASA_2020_set[NASA_2020_set['Month']==1]
+anotated_subset = NASA2020[['lon', 'lat', 'label']].merge(anotated_unique[['lon', 'lat', 'pustynia']], on=['lon', 'lat'], how='inner')
+
+accuracy_score(list(anotated_subset['label']), list(anotated_subset['pustynia']))
+
+precision_score(list(anotated_subset['label']), list(anotated_subset['pustynia']))
+
+recall_score(list(anotated_subset['label']), list(anotated_subset['pustynia']))
+
+def GM_monthly_data(month: int, df: pd.DataFrame) -> Tuple[List[int], float, float, float]:
+
+  NASA_M_2020 = df[df['Month']==month]
+  NASA_month = NASA_M_2020.drop(columns=['lon', 'lat', 'Month', 'SoilT_40_100cm', 'SoilM_100_200cm'])
+  scaler = preprocessing.StandardScaler()
+  scaler.fit(NASA_month)
+  standarized = scaler.transform(NASA_month)
+  st_NASA_month = pd.DataFrame(standarized, columns=NASA_month.columns)
+  gm = GaussianMixture(n_components = 2, n_init = 200, max_iter=200, init_params= 'random_from_data', covariance_type='spherical')
+  gm_result = gm.fit_predict(st_NASA_month)
+
+  gm_list = gm_result.tolist()
+  if (sum(gm_list)/len(NASA2020)) >0.5:
+    for i in range(len(gm_list)):
+        gm_list[i] = 1 - gm_list[i]
+
+  deserts_predicted_rate = round(sum(gm_list)/len(NASA2020), 4)
+  coordinates_labels = pd.DataFrame({'lon': NASA_M_2020['lon'].tolist(), 'lat': NASA_M_2020['lat'].tolist(), 'label': gm_list })
+  anotated_subset = coordinates_labels.merge(
+      anotated_unique[['lon', 'lat', 'pustynia']], on=['lon', 'lat'], how='inner')
+  accuracy_on_anotated = accuracy_score(anotated_subset['pustynia'], anotated_subset['label'] )
+  precision_on_anotated = precision_score(anotated_subset['pustynia'], anotated_subset['label'] )
+
+
+  return gm_list, deserts_predicted_rate, round(accuracy_on_anotated, 3), round(precision_on_anotated, 4)
```

Comment:
Tuple są słabym rozwiązaniem dla typów zwracanych w sytuacji gdy nie jesteśmy pewni co się chce zwracać. Chcemy dołożyć jedną metrykę więcej i zmiana w funkcji rozsadza nam kilka miejsce w kodzie. Bezpieczniejsze są Dicty.

Reply:
Faktycznie dict będzie wygodniejszy: wtedy wystarczy zmiana w jednym miejscy kodu, zamiast w dwóch. Dziękuję, zmienię to.